Subject: Fix RAW photos display with new GCC
From: David Kalnischkies <david@kalnischkies.de>

diff --git a/src/exiv2.cc b/src/exiv2.cc
index 455c8d3..98029cd 100644
--- a/src/exiv2.cc
+++ b/src/exiv2.cc
@@ -16,6 +16,7 @@
 #include <exiv2/image.hpp>
 #include <exiv2/exif.hpp>
 #include <iostream>
+#include <string>
 
 // EXIV2_TEST_VERSION is defined in Exiv2 0.15 and newer.
 #ifndef EXIV2_TEST_VERSION
@@ -1130,9 +1131,9 @@ guchar *exif_get_preview(ExifData *exif, guint *data_len, gint requested_width,
 
 	if (!exif->image()) return NULL;
 	
-	const char* path = exif->image()->io().path().c_str();
+	std::string const path = exif->image()->io().path();
 	/* given image pathname, first do simple (and fast) file extension test */
-	gboolean is_raw = filter_file_class(path, FORMAT_CLASS_RAWIMAGE);
+	gboolean is_raw = filter_file_class(path.c_str(), FORMAT_CLASS_RAWIMAGE);
 	
 	if (!is_raw && requested_width == 0) return NULL;
 
@@ -1232,10 +1233,10 @@ extern "C" guchar *exif_get_preview(ExifData *exif, guint *data_len, gint reques
 	if (!exif) return NULL;
 	if (!exif->image()) return NULL;
 
-	const char* path = exif->image()->io().path().c_str();
+	std::string const path = exif->image()->io().path();
 
 	/* given image pathname, first do simple (and fast) file extension test */
-	if (!filter_file_class(path, FORMAT_CLASS_RAWIMAGE)) return NULL;
+	if (!filter_file_class(path.c_str(), FORMAT_CLASS_RAWIMAGE)) return NULL;
 
 	try {
 		struct stat st;
@@ -1246,9 +1247,9 @@ extern "C" guchar *exif_get_preview(ExifData *exif, guint *data_len, gint reques
 		
 		RawFile rf(exif->image()->io());
 		offset = rf.preview_offset();
-		DEBUG_1("%s: offset %lu", path, offset);
+		DEBUG_1("%s: offset %lu", path.c_str(), offset);
 		
-		fd = open(path, O_RDONLY);
+		fd = open(path.c_str(), O_RDONLY);
 		if (fd == -1)
 			{
 			return NULL;
