--- geeqie-1.6/configure.ac~	2020-11-20 21:36:33.000000000 +0000
+++ geeqie-1.6/configure.ac	2021-04-08 22:29:09.742491746 +0000
@@ -253,7 +253,7 @@
 AC_DEFINE_UNQUOTED([GQ_APP_DIR], "$appdir", [Location of application data])
 AC_SUBST(appdir)
 
-eval "eval gq_bindir=${prefix}/lib/${PACKAGE}"
+eval "eval gq_bindir=${libdir}/${PACKAGE}"
 AC_DEFINE_UNQUOTED([GQ_BIN_DIR], "$gq_bindir", [Location of helper scripts and executables])
 AC_SUBST(gq_bindir)
 
